[Bugs] SMF `nsmf-pdusession` `POST /sm-contexts` panics when UDM discovery returns `200` with an empty `NfInstances` list

## Bug Decription
`SMF` protects `nsmf-pdusession` with OAuth2 middleware, but the create path still contains a request-triggerable panic when upstream `UDM` discovery returns `200 OK` with an empty `NfInstances` array.

The create handler always calls `SendNFDiscoveryUDM()` before continuing:

- `NFs/smf/internal/sbi/processor/pdu_session.go:85`
- `NFs/smf/internal/sbi/processor/pdu_session.go:91`

Inside `SendNFDiscoveryUDM()`, the `nil`-error branch dereferences `result.NfInstances[0]` without a length check:

```go
case nil:
    smfContext.UDMProfile = result.NfInstances[0]
```

Code evidence:

- `NFs/smf/internal/sbi/consumer/nrf_service.go:286`
- `NFs/smf/internal/sbi/consumer/nrf_service.go:299`

Docker validation on March 22, 2026 local time:

1. `SMF` was manually restarted after the earlier `UPI` crash validation, which re-registered it to `NRF` with `nfInstanceId=b5d29d3e-3d12-47e1-b21f-5caae3dc00e4`.
2. A valid `nsmf-pdusession` token was obtained from `NRF`.
3. An authenticated minimal `multipart/related` `POST /nsmf-pdusession/v1/sm-contexts` was sent with create data for `DNN=internet`, `S-NSSAI={sst:1, sd:010203}`, `ServingNetwork=208/93`, and a minimal NAS PDU Session Establishment Request body.
4. `NRF` logged a successful discovery lookup:

```text
GET /nnrf-disc/v1/nf-instances?requester-nf-type=SMF&target-nf-type=UDM -> 200
```

5. `SMF` then panicked with:

```text
panic: runtime error: index out of range [0] with length 0
github.com/free5gc/smf/internal/sbi/consumer.(*nnrfService).SendNFDiscoveryUDM
    /go/src/free5gc/NFs/smf/internal/sbi/consumer/nrf_service.go:299 +0x445
```

6. `smf` returned `500` for `POST /nsmf-pdusession/v1/sm-contexts`.

This is a panic-based DoS on the authenticated `nsmf-pdusession` create path under a realistic upstream-registration failure condition. In this lab the process stayed alive because Gin recovery converted the panic into `500`, but the endpoint remains crashable by repeated request traffic whenever `UDM` discovery returns an empty result.

## To Reproduce
Steps to reproduce the behavior:

1. Ensure `SMF` is registered in `NRF`. In this validation the running `SMF` instance ID was:

```text
b5d29d3e-3d12-47e1-b21f-5caae3dc00e4
```

2. Obtain a valid `nsmf-pdusession` token:

```bash
curl -sS -X POST http://10.100.200.3:8000/oauth2/token \
  -H 'Content-Type: application/x-www-form-urlencoded' \
  --data 'grant_type=client_credentials&nfType=SMF&nfInstanceId=b5d29d3e-3d12-47e1-b21f-5caae3dc00e4&targetNfType=SMF&scope=nsmf-pdusession'
```

3. Send an authenticated minimal `multipart/related` `POST /nsmf-pdusession/v1/sm-contexts` that reaches `HandlePDUSessionSMContextCreate()`. In this validation the request used:

- `supi=imsi-208930000000003`
- `pduSessionId=11`
- `dnn=internet`
- `sNssai={sst:1, sd:010203}`
- `servingNetwork={mcc:208, mnc:93}`
- `anType=3GPP_ACCESS`
- `requestType=INITIAL_REQUEST`
- a minimal NAS PDU Session Establishment Request as `N1SmMsg`

4. Confirm with logs:

```bash
docker logs --tail 50 smf 2>&1
docker logs --tail 30 nrf 2>&1
```

Observed server-side output includes:

```text
2026-03-21T23:50:40.855802676Z [INFO][NRF][GIN] | 200 | 10.100.200.6 | GET | /nnrf-disc/v1/nf-instances?requester-nf-type=SMF&target-nf-type=UDM |
2026-03-21T23:50:40.856780441Z [ERRO][SMF][GIN] panic: runtime error: index out of range [0] with length 0
github.com/free5gc/smf/internal/sbi/consumer.(*nnrfService).SendNFDiscoveryUDM
    /go/src/free5gc/NFs/smf/internal/sbi/consumer/nrf_service.go:299 +0x445
2026-03-21T23:50:40.856800907Z [INFO][SMF][GIN] | 500 | 10.100.200.1 | POST | /nsmf-pdusession/v1/sm-contexts |
```

## Expected Behavior
If `UDM` discovery returns an empty result, `SMF` should return a controlled upstream-failure response, not panic.

Expected behavior:

- empty discovery result -> controlled `5xx` or `ProblemDetails`
- no `result.NfInstances[0]` dereference without `len(...)` validation
- no panic and no stack trace on authenticated create traffic

## Screenshots
N/A

## Environment
**Please, complete the following information:**
- free5GC Version: `v4.2.1`
- Runtime image: `free5gc/smf:v4.2.1`
- Runtime SMF commit hash: `8385c00a`
- Validation date: March 22, 2026 local time (`2026-03-21T23:50:40Z` in logs)
- OS: `Ubuntu 22.04 LTS`
- Kernel version: `5.15.0-88-generic`
- go version: `go1.24.9 linux/amd64`
- C compiler version (optional): `gcc (Ubuntu 11.2.0-19ubuntu1) 11.2.0`

## Trace Files
### Configuration Files
No source patch or persistent configuration change was required for this reproduction. The issue was observed against the stock Docker-based `SMF` runtime for free5GC `v4.2.1`, with the runtime condition that `NRF` returned `200 OK` to UDM discovery but the `NfInstances` array was empty.

### PCAP File
Not attached in this report.

### Log File
Relevant `smf` log excerpt:

```text
2026-03-21T23:50:40.848733224Z [INFO][SMF][PduSess] Receive Create SM Context Request
2026-03-21T23:50:40.848855604Z [INFO][SMF][PduSess] In HandlePDUSessionSMContextCreate
2026-03-21T23:50:40.856780441Z [ERRO][SMF][GIN] panic: runtime error: index out of range [0] with length 0
github.com/free5gc/smf/internal/sbi/consumer.(*nnrfService).SendNFDiscoveryUDM
    /go/src/free5gc/NFs/smf/internal/sbi/consumer/nrf_service.go:299 +0x445
github.com/free5gc/smf/internal/sbi/processor.(*Processor).HandlePDUSessionSMContextCreate
    /go/src/free5gc/NFs/smf/internal/sbi/processor/pdu_session.go:86 +0x658
2026-03-21T23:50:40.856800907Z [INFO][SMF][GIN] | 500 | 10.100.200.1 | POST | /nsmf-pdusession/v1/sm-contexts |
```

Relevant `nrf` log excerpt:

```text
2026-03-21T23:50:40.855468027Z [INFO][NRF][DISC] Handle NFDiscoveryRequest
2026-03-21T23:50:40.855802676Z [INFO][NRF][GIN] | 200 | 10.100.200.6 | GET | /nnrf-disc/v1/nf-instances?requester-nf-type=SMF&target-nf-type=UDM |
```

## System Architecture (Optional)
Docker-based free5GC lab on the internal SBI network:

- source host issuing PoCs: `10.100.200.1`
- NRF: `10.100.200.3:8000`
- SMF: `10.100.200.6:8000`

## Walkthrough (Optional)
The authenticated create path reaches `HandlePDUSessionSMContextCreate()`, which always performs UDM discovery before later request processing. If `NRF` returns `200 OK` with an empty `NfInstances` list, the `nil`-error branch in `SendNFDiscoveryUDM()` dereferences `result.NfInstances[0]` without checking length, producing a deterministic panic.

## Additional Context
- This report is distinct from issue `#894`. Issue `#894` describes a broader `500`-error robustness problem during malformed PDU session flows, while this report identifies a concrete panic sink in `SendNFDiscoveryUDM()` with exact line-level evidence and a specific upstream trigger condition.
- The original test intent was to drive the later `AMF` discovery path with a bogus `ServingNfId`, but the current lab exposed a stronger earlier crash in `UDM` discovery because the `UDM` discovery result was empty first.

--- a/internal/sbi/consumer/nrf_service.go
+++ b/internal/sbi/consumer/nrf_service.go
@@ -296,6 +296,10 @@ func (s *nnrfService) SendNFDiscoveryUDM() (*models.ProblemDetails, error) {
 	case error:
 		return openapi.ProblemDetailsSystemFailure(err.Error()), nil
 	case nil:
+		if len(result.NfInstances) == 0 {
+			logger.ConsumerLog.Warnln("UDM discovery returned empty NfInstances")
+			return nil, openapi.ReportError("UDM discovery returned empty NfInstances")
+		}
 		smfContext.UDMProfile = result.NfInstances[0]
 
 		var client *SubscriberDataManagement.APIClient
@@ -370,6 +374,10 @@ func (s *nnrfService) SendNFDiscoveryServingAMF(smContext *smf_context.SMContext
 		if result.NfInstances == nil {
 			logger.ConsumerLog.Warnln("NfInstances is nil")
 			return nil, openapi.ReportError("NfInstances is nil")
+		}
+		if len(result.NfInstances) == 0 {
+			logger.ConsumerLog.Warnln("AMF NfInstances is empty")
+			return nil, openapi.ReportError("AMF NfInstances is empty")
 		}
 		logger.ConsumerLog.Info("SendNFDiscoveryServingAMF ok")
 		smContext.AMFProfile = result.NfInstances[0]

